lib/src/package_analysis/package_analysis.dart

```diff
@@ -21,50 +30,384 @@ class SummaryCommand extends Command {
+      topPackages.removeWhere(incompatiblePackages.contains);
+      allPackages.removeWhere(incompatiblePackages.contains);
+    } finally {
+      c.close();
+    }
+
+    if (packageCountToAnalyze > allPackages.length) {
+      stderr.writeln(
+          'Provided package count to analyse is too high, analysing all ${allPackages.length} available packages instead.');
+      packageCountToAnalyze = allPackages.length;
+    }
+    final packagesToAnalyse =
+        packageCountToAnalyze > topPackages.length ? allPackages : topPackages;
+
+    // ensures that the indexes up to and including packageCountToAnalyze do not
+    // lose their order when ordered lexicographically
+    String formatIndexForLogfile(int index) {
+      final numberOfZeroes = packageCountToAnalyze.toString().length;
+      return (index + 1).toString().padLeft(numberOfZeroes, '0');
+    }
+
+    final pool = Pool(resourceCount);
+
+    await Future.wait(packagesToAnalyse
+        .getRange(0, packageCountToAnalyze)
+        .mapIndexed((targetPackageIndex, targetPackageName) async {
+      await pool.withResource(() async {
+        print(
+            '${DateTime.now().toIso8601String()} Analyzing package $targetPackageName (${targetPackageIndex + 1}/$packageCountToAnalyze)...');
+
+        late final Process process;
+        try {
+          process = await Process.start(
+            Platform.resolvedExecutable,
+            [
+              'run',
+              packageAnalysisFilePath,
+              lbcAnalysisCommandName,
+              cachePath,
+              targetPackageName,
+            ],
+          );
+        } catch (e) {
+          stderr.writeln(
+              'Failed to start analysis process on package $targetPackageName with error $e.');
+          return;
+        }
+
+        try {
+          final logPathPrefix = path.join(logPath,
+              '${formatIndexForLogfile(targetPackageIndex)} $targetPackageName');
+
+          // wait for the analysis to complete or time out after 10 minutes
+          final exitCode =
+              await process.exitCode.timeout(const Duration(minutes: 10));
+
+          // capture all output
+          final out = await process.stdout.toList();
+          final err = await process.stderr.toList();
+
+          // do not write empty log files
+          if (out.isNotEmpty) {
+            final stdoutLog = File('$logPathPrefix stdout.txt').openWrite();
+            out.forEach(stdoutLog.add);
+            await stdoutLog.flush();
+            await stdoutLog.close();
+          }
+          if (err.isNotEmpty) {
+            final stderrLog = File('$logPathPrefix stderr.txt').openWrite();
+            err.forEach(stderrLog.add);
+            await stderrLog.flush();
+            await stderrLog.close();
+          }
+
+          if (exitCode != 0) {
+            stderr.writeln(
+                'Analysis of package $targetPackageName completed with a non-zero exit code.');
+          }
+        } catch (e) {
```

Comment:
You might want to do `} on TimeoutException {` and handle this separately, because then you can display a message specific to timeout issues..

Also, catching all exceptions/errors can hide internal error that you then fail to discover.